packages/cli-hydrogen/src/cli/services/deploy.ts

```diff
@@ -0,0 +1,97 @@
+import {DeployConfig, ReqDeployConfig} from './deploy/types.js'
+import {createDeployment, healthCheck, uploadDeployment} from './deploy/upload.js'
+import {buildTaskList} from './build.js'
+import {validateProject, fillDeployConfig} from './deploy/config.js'
+import {environment, system, ui} from '@shopify/cli-kit'
+
+interface TaskContext {
+  config: ReqDeployConfig
+  deploymentID: string
+  assetBaseURL: string
+  previewURL: string
+}
+
+const isUnitTest = environment.local.isUnitTest()
+
+export async function deployToOxygen(_config: DeployConfig) {
+  const backoffPolicy = [5, 10, 15, 30, 60]
+
+  await validateProject(_config)
+
+  /* eslint-disable require-atomic-updates */
+  const tasks: ui.ListrTask<TaskContext>[] = [
+    {
+      title: '📝 Getting deployment config',
+      task: async (ctx, task) => {
+        ctx.config = await fillDeployConfig(_config)
+        task.title = '📝 Deployment config parsed'
+      },
+    },
+    {
+      title: '✨ Creating a deployment',
```

Comment:
```suggestion
      title: '💡 Initializing deployment',
```
I'm going with [initialize](https://www.merriam-webster.com/dictionary/initialize) over [initiate](https://www.merriam-webster.com/dictionary/initiate), I think it's the more specific meaning. Also we should drop the article here.

Alternate emoji ideas: 🌱 🧩 ⚡